lib/shape.ml

```diff
@@ -60,27 +62,34 @@ let line ?x1 ?y1 x2 y2 =
   | Some x, Some y -> Line { a = { x; y }; b = { x = x2; y = y2 } }
   | _ -> Line { a = { x = 0; y = 0 }; b = { x = x2; y = y2 } }
 
-let translate dx dy shape =
+let complex arg = 
+  match arg with 
+  | [single] -> Complex [single] 
+  | _ :: _ -> Complex arg
+  | [] -> failwith "invalid complex shape"
+
+let rec translate dx dy shape =
   match shape with
   | Circle circle ->
-      Circle { circle with c = { x = circle.c.x + dx; y = circle.c.y + dy } }
-  | Rectangle rectangle ->
-      Rectangle
-        {
-          rectangle with
-          c = { x = rectangle.c.x + dx; y = rectangle.c.y + dy };
-        }
-  | Ellipse ellipse ->
-      Ellipse
-        { ellipse with c = { x = ellipse.c.x + dx; y = ellipse.c.y + dy } }
-  | Line line ->
-      Line
-        {
-          a = { x = line.a.x + dx; y = line.a.y + dy };
-          b = { x = line.b.x + dx; y = line.b.y + dy };
-        }
-
-let scale factor s =
+    Circle { circle with c = { x = circle.c.x + dx; y = circle.c.y + dy } }
+| Rectangle rectangle ->
+    Rectangle
+      {
+        rectangle with
+        c = { x = rectangle.c.x + dx; y = rectangle.c.y + dy };
+      }
+| Ellipse ellipse ->
+    Ellipse
+      { ellipse with c = { x = ellipse.c.x + dx; y = ellipse.c.y + dy } }
+| Line line ->
+    Line
+      {
+        a = { x = line.a.x + dx; y = line.a.y + dy };
+        b = { x = line.b.x + dx; y = line.b.y + dy };
+      }
+  | Complex complex -> Complex (List.map (translate dx dy) complex)
```

Comment:
Can you rename `complex` here to something more readily understandable? (e.g. shapes)